drivers/hmis/app/models/hmis/hud/processors/disability_group_processor.rb

```diff
@@ -9,7 +9,11 @@ class DisabilityGroupProcessor < Base
     def process(field, value)
       if field_mapping.key?(field)
         disability_type, disability_field, enum_type = field_mapping[field]
-        disability_value = attribute_value_for_enum(enum_type, value)
+        disability_value = if enum_type.values.values.map(&:value).map(&:to_s).include?(value)
```

Comment:
Not sure what you will think of this, but this is the hacky workaround to the fact that the processor expects these values to come from enums, but in the new form they are booleans.

but, as mentioned above, it doesn't work for substance abuse disorder where 1 means "alcohol use disorder (only)" and not simply "yes".

Reply:
Part of me thinks we should just make `attribute_value_for_enum` more forgiving (like below), so it can flexibly allow "real" values alongside enum values. Then we don't need to special-case it in each form processor any time we have an external form doing something like this. 
I _think_ this would be appropriate, because the form processor is already forgiving in the way it ingests CDEDs (eg bool can be `0`,`'0'`,`false`), so we have the concept already.
What do you think? 

```
attribute_value_for_enum(Types::HmisSchema::Enums::Hud::NoYesReasonsForMissingData, 'YES') => 1
attribute_value_for_enum(Types::HmisSchema::Enums::Hud::NoYesReasonsForMissingData, '1') => 1
attribute_value_for_enum(Types::HmisSchema::Enums::Hud::NoYesReasonsForMissingData, 1) => 1
```

Reply:
@gigxz i think it sounds sensible to me that `attribute_value_for_enum` should be flexible, but I'm not sure exactly how it should be implemented; how would that method know whether an int or a string should be returned, other than by checking the values of the enum type provided to see if they are integers? I've taken a stab at that here but I think it's kind of ugly.

https://github.com/greenriver/hmis-warehouse/pull/4768/commits/47ea7fe9373e822d01c9d2ee2a8a90b5153bec3e

Reply:
This also still wouldn't quite work for substance abuse disorder. Maybe the right path forward is to make 2 new accepted field mappings `alcoholUseDisorder` and `drugUseDisorder` that collect 2 different booleans, and then collect that onto the same disability record?

Reply:
A cleaner approach might be to create an alternate method like `value_for` but that accepts a key or value (eg `'1',1,'YES'`, and call that instead of value_for.